pontoon/contributors/utils.py

```diff
@@ -478,39 +488,70 @@ def get_contribution_timeline_data(user, contribution_type=None, day=None):
         contributions_qs = contributions_map[contribution_type]
         contribution_data = get_project_locale_contribution_counts(contributions_qs)
 
-        c_count = sum([value["count"] for _, value in contribution_data.items()])
-        p_count = len(contribution_data)
-
-        if c_count == 0:
-            continue
-
-        # Generate localizaton URL and add it to the data dict
+        title = ""
+        for month, data in contribution_data.items():
+            total_count = 0
+            projects = set()
+            for key, info in data.items():
+                total_count += info["count"]
+                # Count number of unique project/locale pairs
+                projects.add(key)
+
+            # Generate title for the localizations belonging to the same contribution type
+            p_count = len(projects)
+            if contribution_type == "user_translations":
+                title = f"Submitted { intcomma(total_count) } translation{ pluralize(total_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+            elif contribution_type == "user_reviews":
+                title = f"Reviewed { intcomma(total_count) } suggestion{ pluralize(total_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+            elif contribution_type == "peer_reviews":
+                title = f"Received review for { intcomma(total_count) } suggestion{ pluralize(total_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+
+            # Add title as a field
+            if month not in contribution_data:
+                contribution_data[month] = {}
+            contribution_data[month]["title"] = title
```

Comment:
We should avoid such mutations of existing dicts, otherwise we need to do things like `if field != "title":`, which aren't pretty.

Instead, we should be able to store title directly to the new `contributions` dict.